frontend/src/app/ml-models/ml-model-form/ml-model-form.component.ts

```diff
@@ -309,10 +321,12 @@ export class MlModelFormComponent implements OnInit {
       labelField.key.setValue('');
     }
 
-    if (label.name) {
-      // set source automatically in the form based on label selected.
-      labelField.source.setValue(label.source);
+    // default to Google Analytics as the source if no first party data is used.
+    if (!this.value('usesFirstPartyData')) {
+      labelField.source.setValue(Source.GOOGLE_ANALYTICS);
+    }
 
+    if (label.name) {
```

Comment:
There are repeated conditions (lines 212 and 319) for the existence of label.name. Is this intentional?

Reply:
So, the logic is a bit different. One is just looking to see if a label name is set and if so, it's associating all the other properties of it from all available variables (`get label()`). `refreshLabel()` is taking into consideration changes that need to happen when fields are changed in the form and looking at only the labels that should be available (`get labels()`) at that moment and ensuring that the current selected label is in that set. So, if you change the source of the label in the UI it should unset the label name and key fields. If you select your current label as a feature it should also unset it as the label as well.

Though, I did miss something. I should have also checked the source in this logic as well. I've updated that.

Reply:
Good find. There were actually quite a bit of issues with the order of things happening here that I completely missed. I've updated a good deal of this and tested quite thoroughly to ensure it's working as intended now.

To summarize the changes I've made:
1. The source is updated first in the event you remove first party data as an available data source.
2. The labels are limited based on this source and so are now checked after this source is updated. As part of this if the selected label is no longer available it's unset.
3. Only after the above checks and changes take place does the label get pulled and checked further.

I've also added a feature refresh as well. Realized that if you changed the data source to remove 1P data then the 1P features remained which would be a problem later.